sigma/plugins/roles/setcolor.py

```diff
@@ -0,0 +1,62 @@
+import discord
+from sigma.core.permission import check_man_roles
+from sigma.core.rolecheck import matching_role
+
+async def setcolor(cmd, message, args):
+    # if no args, send help message
+    if not args:
+        await message.channel.send(cmd.help())
+        return
+
+    # checks permissions
+    if not check_man_roles(message.author, message.channel):
+        out_content = discord.Embed(type='rich', color=0xDB0000, title='⛔ Error')
+        out_content.add_field(name='Insufficient Permissions', value='You require the Manage Roles permission.')
+        await message.channel.send(None, embed=out_content)
+        return
+
+    color = args[0].lstrip('#').upper()
+    # check if color is a proper format
+    if not is_valid_hex(color):
+        out_content = discord.Embed(type='rich', color=0xFF9900, title='⚠ Error')
+        out_content.add_field(name='Invalid Input', value='A valid hexadecimal color code has numbers from 0–9 and letters from A–F, and has six characters.')
+        await message.channel.send(None, embed=out_content)
+        return
+
+    # check if role exists
+    exists = matching_role(message.guild, color)
+    if not exists:
+        discord_colour = discord.Colour(int(color, 16))
+        target_role = await message.guild.create_role(name=color, colour=discord_colour, reason="Color role")
+    else:
+        target_role = exists
+
+    target = message.author if not message.mentions else message.mentions[0]
+
+    # get user's roles, remove old role, append new role, and set user's roles
+    list_of_roles = target.roles
+
+    for index, role in enumerate(list_of_roles):
+        if is_valid_hex(role.name):
+            await list_of_roles[index].delete()
+
+    list_of_roles.append(target_role)
+
+    await target.edit(roles=list_of_roles)
```

Comment:
see `add_roles` in https://discordpy.readthedocs.io/en/latest/api.html#discord.Member.add_roles

Reply:
Is there a `remove_roles`?

Reply:
https://discordpy.readthedocs.io/en/latest/api.html#discord.Member.remove_roles

Reply:
Done.